### DISCUSSION TITLE
Question: Decomposing path to cell paths?

### DISCUSSION THREAD

**Discussion by akabbeke** [uber/h3#928] [Q&A]
Does there exist an efficient way to decompose a straight line between two location into the individual line segments which cross each cell at a given resolution?

Apologies if this is the wrong place to ask this.

**Comment by akabbeke** [comment]
The solution I was considering was to try a kind of binary search. Find where the line intersects the cell boundary for the first and last cells. If the intersections are at the same location then we know they are neighbours and the line passed through a shared boundary edge. If not then find the midpoint between the two points and repeat the process until an unbroken line exists between the original start and end points. 

I’m fairly certain the above approach works, but I’m not sure if it is the most efficient approach or if there exists a better method.

**Comment by dfellis** [comment]
This should be a discussion, not an issue with the library. I am going to try to convert it, but this is my first time doing so.

**Comment by dfellis** [comment]
On the subject of your exact question, can you clarify why you want to do what you're describing? Eg, why doesn't [`gridPathCells`](https://h3geo.org/docs/api/traversal#gridpathcells) cover what you need?
